tests/scripts/all.sh

```diff
@@ -531,8 +531,7 @@ cleanup
 cp "$CONFIG_H" "$CONFIG_BAK"
 scripts/config.pl full
 scripts/config.pl unset MBEDTLS_MEMORY_BACKTRACE # too slow for tests
-CC=clang cmake -D CMAKE_BUILD_TYPE:String=Check -D ENABLE_TESTING=On .
-make
+make CC='clang' CFLAGS='-Werror -O2'
```

Comment:
Was it intentional to change from cmake to make here?

Reply:
I guess: Is it for ease to specify ```CFLAGS```.

Reply:
@Patater It was not intentional to do it in this commit (otherwise it would have been motivated in the commit message - probably messed up something while trying to clean up history after rebasing), but it was intentional. The motivation is that the first build uses cmake, so by using make here we test both build systems in the first two builds (just like we do first build with gcc and this one with clang). This is not a very strong motivation, but since the choice of `cmake` vs `make` is otherwise quite arbitrary here, why not?

@gilles-peskine-arm I'll fix the message.

@mazimkhan That was not really my motivation here, as `Check` is supposed to set exactly the flags we want anyway.